pages/blog/miniseries-part_1.md

```diff
@@ -0,0 +1,227 @@
+
+The main point of the **game server** will be to simulate the interaction of the player. This means it will simulate the following events: when players join the game server, pick up items in game, uses the chat, hit one another, and eventually disconnect.
+
+The backend **processor** will be subscribing to these events to process them. In our example it will simply save the received data directly to a database.
+
+I will not get into the specifics of the stack for this system yet since it have no effect on the writing of the API documents for the two applications.
+
+# Designing the API's with AsyncAPI
+I always use the [design first principle](https://apisyouwonthate.com/blog/api-design-first-vs-code-first), even when we are talking about internal systems, which means describing the two applications, **game server** and **processor** using the AsyncAPI specification before starting with the implementation.
+
+Using AsyncAPI to define an internal system is not entirely apparent how to do, since AsyncAPI is build to define behavior from the perspective of the external user. I will be clarifying this a bit more later on with actual examples, but if you want to go deeper into this, or alternatively read Nic Townsend's post about [Demystifying the Semantics of Publish and Subscribe](https://www.asyncapi.com/blog/publish-subscribe-semantics). 
+
+## The game server
+I always start basic, and first define all the different channels for which the **game server** should produce events over.
+
+```yaml
+asyncapi: 2.0.0
+info: 
+  title: "Game server"
+  version: "0.0.1"
+channels: 
+  game/server/{serverId}/events/player/{playerId}/item/{itemId}/pickup:
+    description: Channel used when a player picks up an item in-game
+  game/server/{serverId}/events/player/{playerId}/connect:
+    description: Channel used when a player joins (connect to) the game server
+  game/server/{serverId}/events/player/{playerId}/disconnect:
+    description: Channel used when a player leaves (disconnects from) the game server
+  game/server/{serverId}/events/player/{playerId}/chat: 
+    description: Channel used when a player writes something in chat
+  game/server/{serverId}/events/player/{playerId}/hit: 
+    description: Channel used when a player hit another player in-game
+```
+
+Channels must be defined as a [RFC 6570 URI template](https://www.asyncapi.com/docs/specifications/2.0.0#channelsObject), regardless of how the underlying broker . The way I like to structure my channels, is to utilize parameters to separate the action from information about the event, so it describes, on what server the event was performed `{serverId}`, by what player `{playerId}` and in case of **pickup**, what item `{itemId}` gets picked up. For the last part of the channel we describe what the event was, **pickup**, **connect**, **disconnect**, etc.
```

Comment:
`regardless of how the underlying broker .` looks like you cut the sentence in the middle 🤔